crates/core/src/bucket_priority.rs

```diff
@@ -0,0 +1,89 @@
+use core::ops::RangeInclusive;
+
+use serde::{de::Visitor, Deserialize};
+use sqlite_nostd::ResultCode;
+
+use crate::error::SQLiteError;
+
+#[repr(transparent)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct BucketPriority(i32);
+
+impl BucketPriority {
+    pub fn may_publish_with_outstanding_uploads(self) -> bool {
+        self == BucketPriority::HIGHEST
+    }
+
+    pub const HIGHEST: BucketPriority = BucketPriority(0);
+    pub const LOWEST: BucketPriority = BucketPriority(3);
```

Comment:
Can we remove the restriction on the lowest priority here? The only place I see it being used apart from validation is to detect the "sync everything" case for sync_local, which can also be done by just passing in a null parameter.

I'm thinking that we can leave it up to the service to specify the allowed range of priorities, and have 0 as the only special case on the client.